File: internal/dto/account.go
package dto

// AccountResponse is returned to the account owner. Currency is fixed to RUB by the service/database rules.
type AccountResponse struct {
	ID            int64  `json:"id"`
	AccountNumber string `json:"account_number"`
	Balance       string `json:"balance"`
	Currency      string `json:"currency"`
	IsBlocked     bool   `json:"is_blocked"`
	Status        string `json:"status"`
	ClosedAt      string `json:"closed_at,omitempty"`
	CreatedAt     string `json:"created_at"`
}

type CloseAccountResponse struct {
	ID            int64  `json:"id"`
	AccountNumber string `json:"account_number"`
	Status        string `json:"status"`
	ClosedAt      string `json:"closed_at"`
	Message       string `json:"message"`
}

type DepositRequest struct {
	Amount string `json:"amount"`
}

type WithdrawRequest struct {
	Amount  string `json:"amount"`
	MFACode string `json:"mfa_code"`
}

type PredictBalanceResponse struct {
	AccountID               int64  `json:"account_id"`
	Days                    int    `json:"days"`
	CurrentBalance          string `json:"current_balance"`
	ExpectedIncome          string `json:"expected_income"`
	ExpectedExpense         string `json:"expected_expense"`
	ScheduledCreditPayments string `json:"scheduled_credit_payments"`
	PredictedBalance        string `json:"predicted_balance"`
	AverageDailyIncome      string `json:"average_daily_income"`
	AverageDailyExpense     string `json:"average_daily_expense"`
	StatisticsPeriodDays    int    `json:"statistics_period_days"`
}

File: web/src/types/account.ts
export type AccountResponse = {
  id: number
  account_number: string
  balance: string
  currency: string
  is_blocked: boolean
  status: string
  closed_at?: string
  created_at: string
}

export type CloseAccountResponse = {
  id: number
  account_number: string
  status: string
  closed_at: string
  message: string
}

export type PredictBalanceResponse = {
  account_id: number
  days: number
  current_balance: string
  expected_income: string
  expected_expense: string
  scheduled_credit_payments: string
  predicted_balance: string
  average_daily_income: string
  average_daily_expense: string
  statistics_period_days: number
}
